### ISSUE TITLE
Incompatibility with Razer BlackWindow Chroma V2 Keyboard

### ISSUE THREAD

**Issue by DemolitionDerby** [cxong/cdogs-sdl#685]
Hi,

For some reason, the game does not seem to like my keyboard lol

Issue: When my keyboard is plugged in, the character will automatically walk forward and there is no way to stop its movement.

When I unplug my keyboard and try with a controller, it works perfectly.

I have also tried closing the Razer Control Panel to see if it could be the cause but it didn't change anything.

I know for a fact that this is probably an isolated and wierd issue to resolve but please let me know if there is any additional information I can provide if the issue is going to be considered.

Thanks!

**Comment by cxong** [comment]
That keyboard sounds like a normal keyboard so I don't see why it wouldn't work
Can you try a keyboard testing program like https://www.keyboardtester.com/ to see if it's not giving any weird inputs

**Comment by DemolitionDerby** [comment]
Just tested, everything seems fine on the Tester, no "stuck" keys or anything.

I have a second computer, also with Razer keyboard and mouse (different models) and the I have the same problem.

I just tested something else, even if my keyboard is plugged in, if I start the game using a controller, it will work correctly.

The keyboard input seems to fail when I get to the screen that says "Press Fire to continue", with a controller (XBox), I can just press A and it will move on but at that point, every keyboard input seems to fail.

I can't press "Enter" to continue of the Fire key, it just dosent work.

I have also tried with another keyboard that isn't Razer and I seem to be having the same issue, on 2 seperate computers lol

**Comment by cxong** [comment]
Check your key bindings - by default fire is `x` https://github.com/cxong/cdogs-sdl/blob/master/src/cdogs/config.c#L660
This can be changed in the options menu
The "Press Fire to continue" is an input device selection screen - if you press `x` then keyboard 1 is associated with the player (note that Enter doesn't work here because Enter is not fire)
If you press `A` on the controller, the controller now controls that player, not the keyboard

After this point only player controls work; Enter doesn't do anything

**Comment by DemolitionDerby** [comment]
Yeah, I tried it doesn't work, past that point even if I press the arrows on my keyboard to navigate in the menu, it doesn't work either.

Thank you for replying and trying to assist me with this I don't think this is something that you will be able to fix as this random issue does seem to be a really isolated problem on my side (2 different computers? How? xD)

I think I'll fork the code and try to debug my way out of it and if I manage to get it fixed, I'll send a pull request :)
